### PR TITLE
feat: audience subscribers dynamic sidebar nav menu

### PR DESCRIPTION
**Pull request by yacosta738**
This pull request introduces a dynamic sidebar navigation system, along with supporting tests, documentation, and updates to the project dependencies. The changes focus on creating a flexible and configurable navigation component (`AppSidebar.vue`) with advanced features such as nested menus, conditional visibility, access control, and active state management. Additionally, tests and documentation ensure the system meets the specified requirements and integrates seamlessly with the existing codebase.

#### Dynamic Sidebar Navigation Implementation:
* **Design and Requirements Documentation**: Added `.kiro/specs/dynamic-sidebar-navigation/design.md` and `.kiro/specs/dynamic-sidebar-navigation/requirements.md` to outline the architecture, components, and functional requirements of the dynamic sidebar navigation system. [[1]](diffhunk://#diff-80098a03a622a79918d4cca0d1517688762fecc27ab6a401befe73216b0565e1R1-R287) [[2]](diffhunk://#diff-5eb2280d52aeab8a23614e0b4b33fb3337f76af7d555ea424428a114d3efbc81R1-R79)
* **Implementation Plan**: Added `.kiro/specs/dynamic-sidebar-navigation/tasks.md` detailing the step-by-step implementation of the sidebar, including recursive components, active state management, and error handling.
* **New Components**: Created `AppSidebar.vue` and `AppSidebarItem.vue` with TypeScript interfaces and utility functions for visibility, access control, and active state detection. These components support unlimited nesting and integrate with existing UI elements like tooltips and collapsibles.

#### Testing and Validation:
* **Unit and Integration Tests**: Added tests in `client/apps/web/src/config/navigation.test.ts` to validate navigation configuration, including visibility, access control, and active state management. Mocked the authentication store to simulate various user roles and permissions.
* **Test Results**: All 145 tests across utility functions, components, and integration scenarios are passing, ensuring robust functionality and edge case handling.

#### Dependency Updates:
* **Logger Package**: Added `@hatchgrid/logger` as a new dependency in `client/apps/web/package.json` to enhance logging capabilities.

#### Miscellaneous:
* **Subscriber Page**: Added a placeholder component `Subscribers.vue` under `client/apps/web/src/audience/subscriber` to support the new "Audience" navigation item.
* **Improved Test Hook**: Updated `.kiro/hooks/test-check-build.kiro.hook` to clarify instructions for running tests, checks, and builds, ensuring better developer guidance.

### CODE REVIEW

File: client/apps/web/src/layouts/components/sidebar/composables/useNavigationState.ts [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+		},
+		{ immediate: true },
+	);
+
+	const debouncedUpdate = debounce((newItems: unknown) => {
+		navigationStateManager.updateRoute(
+			route.path,
+			newItems as AppSidebarItem[],
+		);
+	}, 100);
```
[the remark this answers is not in the document]
Comment (yacosta738) [reply client/apps/web/src/layouts/components/sidebar/composables/useNavigationState.ts:180]: /**
 * Creates a debounced function that delays invoking the provided function until after
 * a specified wait time has elapsed since the last time the debounced function was called.
 * Optionally, the function can be invoked immediately on the leading edge instead of the trailing.
 *
 * @template T - The type of the function to debounce.
 * @param {T} func - The function to debounce.
 * @param {number} wait - The number of milliseconds to delay.
 * @param {boolean} [immediate] - If `true`, triggers the function on the leading edge, instead of the trailing.
 * @returns {DebouncedFunction<T>} - Returns the new debounced function.
 */
type DebouncedFunction<T extends (...args: unknown[]) => unknown> = (
	...args: Parameters<T>
) => void;

export function debounce<T extends (...args: unknown[]) => unknown>(
	func: T,
	wait: number,
	immediate?: boolean,
): DebouncedFunction<T> {
	let timeout: ReturnType<typeof setTimeout> | null;

	return function (this: ThisParameterType<T>, ...args: Parameters<T>) {
		if (timeout) {
			clearTimeout(timeout);
		}

		if (immediate && !timeout) {
			func.apply(this, args);
		}

		timeout = setTimeout(() => {
			timeout = null;
			if (!immediate) {
				func.apply(this, args);
			}
		}, wait);
	};
}

**Review by yuniel-acosta — APPROVED** [review]
